pycargoebuild/__main__.py

```diff
@@ -397,6 +397,8 @@ def repack_crates(tar_out: tarfile.TarFile,
                                      delete=False) as outf:
         try:
             if args.input is not None:
+                st = os.stat(args.input.fileno())
+                os.chown(outf.fileno(), st.st_uid, st.st_gid)
                 # typeshed is missing fd support in shutil.copymode()
                 # https://github.com/python/typeshed/issues/9288
                 shutil.copymode(args.input.fileno(),
```

Comment:
Since you're doing a `stat()` now, let's replace this with `os.chmod()` as well.

Reply:
Done. This lets us totally get rid of the shutil import, too...